enhancements/results-forwarding.md

```diff
@@ -0,0 +1,379 @@
+**resultserver** then becomes subject to the evidence forwarding provider and will be renamed
+**evidence-persistor**.
+
+Forwarder implementations would be configured via the `ScanSetting` and they'd be called "providers".
+For backwards compatibility, we'd have defaults that would point to the current mode of operation
+of the Compliance Operator. e.g. a `pvc` provider for evidence storage.
+
+For more detail on the proposed API, see the *API Extensions* section.
+
+### User Stories
+
+* As an Site Reliability Engineer managing multiple clusters, I'd like to run compliance
+  scans on my fleet in such a way that uses the minimal amount of resources.
+
+* As a Site Reliability Engineer managing multiple clusters, I'd like to have
+  one place to view the compliance stance of my whole fleet.
+
+### API Extensions
+
+Given the scope of the change, we'd upgrade the ScanSetting's version to `v1alpha2`.
+
+A sample `ScanSetting` object with Compliance Operator's current mode of operation would look as follows:
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha2
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+generateResultCRDs: true  # Adds CRD creator and sets forwarder config to send info there
+resultForwarding:
+  ... # Potential configuration goes here.
+evidenceForwarding:
+  provider: pvc
+  pvc: # configuration goes here. This would be the same as `rawResultStorage` in ScanSetting v1alpha1
+    pvAccessModes:
+      - ReadWriteOnce 
+    rotation: 3 
+    size: 1Gi
+roles:
+- worker 
+- master 
+scanTolerations: 
+  default:
+  - operator: Exists
+  schedule: 0 1 * * * 
+```
+
+To forward results, we'd implement a `grpc` provider. A sample would look as follows:
+
+```yaml
+apiVersion: compliance.openshift.io/v1alpha2
+kind: ScanSetting
+metadata:
+  name: default
+  namespace: openshift-compliance
+generateResultCRDs: false # no CRD-creator container will be provisioned
+storeEvidenceInPVC: false # no PVC store container will be provisioned
+evidenceStorage: null # No storage options are needed
+resultForwarding:
+  provider: grpc
+  grpc:
+    version: v1  # we'll start at v1
+    endpoint: <URL to GRPC service (could be in-cluster or outside of it)>
+    tls:
+      dynamic: 
+      caSecretName: <ref to CA cert/key pair for TLS> . # This is mandatory
+      serverSecretName: <ref to server cert/key pair secret for mTLS>
+      clientSecretName: <ref to client cert/key pair secret for mTLS> # Mandatory if mtls authentication is used
+    authentication:
+      type: <mtls or token>
+      token:  # May be nil if mtls is chosen
+        tokenSecretName: <ref to secret containing token>
+    extraMetadata:  # defined as a map[string]string
+      cluster_name: foo
+      cluster_type: bar
+      randomKey: random value
+...
+```
+
+Note that `generateResultCRDs` is set to `false`, as well as `storeEvidenceInPVC`
+
+```yaml
+evidenceForwarding:
```

Comment:
Is this meant to be a separate example from the GRPC example above? It looks like it will be possible to send evidence one place and results to another.

I was under the impression one provider (GPRC) would forward all information (results and evidence) to the same place.

Reply:
After looking at the evidence arguments, and the size of the ARF files, I can see how sending evidence over gRPC is going to be problematic.